pkg/operator/ceph/config/conditions.go

```diff
@@ -117,28 +117,32 @@ func translatePhasetoState(phase cephv1.ConditionType) cephv1.ClusterState {
 	}
 }
 
-// Updating the status of Progressing, Updating or Upgrading to False once cluster is Ready
-func checkConditionFalse(context *clusterd.Context, namespace, name string) {
+// MakeConditionFalse Updates the status of Progressing, Updating or Upgrading to False once cluster is Ready
+func MakeConditionFalse(context *clusterd.Context, namespace, name string) {
 	tempConditionList := []cephv1.ConditionType{cephv1.ConditionUpdating, cephv1.ConditionUpgrading, cephv1.ConditionProgressing}
-	var tempCondition cephv1.ConditionType
+	var tempCondition []cephv1.ConditionType
 	for _, conditionType := range tempConditionList {
 		if conditionMap[conditionType] == v1.ConditionTrue {
-			tempCondition = conditionType
+			tempCondition = append(tempCondition, conditionType)
 		}
 	}
-	reason := ""
-	message := ""
-	if tempCondition == cephv1.ConditionUpdating {
-		reason = "UpdateCompleted"
-		message = "Cluster updating is completed"
-	} else if tempCondition == cephv1.ConditionUpgrading {
-		reason = "UpgradeCompleted"
-		message = "Cluster upgrading is completed"
-	} else {
-		reason = "ProgressingCompleted"
-		message = "Cluster progression is completed"
+	for _, conditionType := range tempCondition {
+		reason := ""
+		message := ""
+		if conditionType == cephv1.ConditionUpdating {
+			reason = "UpdateCompleted"
+			message = "Cluster updating is completed"
+		} else if conditionType == cephv1.ConditionUpgrading {
+			reason = "UpgradeCompleted"
+			message = "Cluster upgrading is completed"
+		} else if conditionType == cephv1.ConditionProgressing {
+			reason = "ProgressingCompleted"
+			message = "Cluster progression is completed"
+		}
+		if reason != "" {
+			ConditionExport(context, namespace, name, conditionType, v1.ConditionFalse, reason, message)
```

Comment:
With this PR would you mind renaming `setCondition()` to `SetCondition()` and remove the func `ConditionExport()`? It doesn't seem necessary to have that wrapper around `setCondition()`

Reply:
It certainly is not necessary, but I made the wrapper around `setCondition()` is to make the code cleaner. For eg. in the `controller.go` file, if we need to set the Progressing Condition with this wrap around I can just write
`config.ConditionExport(c.context, namespace, name,
	cephv1.ConditionProgressing, v1.ConditionTrue, "ClusterProgressing", message)`

If I were to remove the wrap around then it would be something like
`config.SetCondition(c.context, namespace, name, cephv1.Condition{
		Type:    conditionType,
		Status:  status,
		Reason:  reason,
		Message: message,
	})
`
And this would be repeating everywhere the condition needs to be set. So instead of doing this I thought it would be nice to have that wrap around. @travisn

Reply:
What about keeping the args to ConditionExport(), but renaming it? The method internally can create the Condition struct. For example:
```
func SetCondition(context *clusterd.Context, namespace, name string, conditionType cephv1.ConditionType, status v1.ConditionStatus, reason, message string) {
	condition := cephv1.Condition{
		Type:    conditionType,
		Status:  status,
		Reason:  reason,
		Message: message,
	})
        // either call setCondition() if it's still useful to have the separate function, or combine that method with this one
...
}
```

Reply:
Done, thanks :)